tdrs-backend/tdpservice/reports/views.py

```diff
@@ -2,14 +2,41 @@
 import logging
 
 from rest_framework import mixins, viewsets
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from ..users.permissions import CanDownloadReport, CanUploadReport
 
-from ..users.permissions import CanUploadReport
-from .models import User
+from .models import User, ReportFile
 from .serializers import ReportFileSerializer
 
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
+class GetYearList(APIView):
+    """Get list of years for which there are reports."""
+
+    query_string = False
+    pattern_name = "report-list"
+    permission_classes = [CanDownloadReport]
+
+    def get(self, request, **kargs):
+        """Handle get action for get list of years there are reports."""
+
+        user = request.user
+        is_ofa_admin = user.groups.filter(name="OFA Admin").exists()
+
+        if ((not kargs.get('stt') and is_ofa_admin)
+                or (not user.stt and not is_ofa_admin)):
+            return Response(status=422)
```

Comment:
```suggestion
```
If you take my suggestion for the changes to `CanDownloadReport` this code won't be reachable anymore, users would instead get a 403 Unauthorized for these situations.